frontend/package.json

```diff
@@ -6,6 +6,7 @@
     "@material-ui/core": "3.7.1",
     "@material-ui/icons": "^3.0.1",
     "@types/js-yaml": "^3.11.2",
+    "aws-sdk": "^2.393.0",
```

Comment:
Please see my comments on https://github.com/kubeflow/pipelines/pull/766

About abstracting away the object store using Minio Client. Let's continue the discussion on https://github.com/kubeflow/pipelines/pull/766 and come back to this PR subsequently.

Reply:
@vicaire I think reuse minio client is good point here. But I am a little bit confused about 

https://github.com/kubeflow/pipelines/blob/eec8099e085332d49c59c3503f7a683fa565351b/frontend/server/server.ts#L35-L41

How to dynamically inject right environment variables or configs based on kubeflow platform?

Reply:
I think maybe I need to attach envs to ml-pipeline-ui pod and read env from it?

Reply:
@rileyjbauer 

Adding Riley to get his opinion on a possible approach. 

Instead of using the environment, could we just have the frontend binary expose parameters that determine the configuration for MinioClient? (S3, GCS, in-cluster MinioServer, etc.).

Reply:
Seems minioClient here is only used to fetch data for artifact view?  
Sounds like a different story as backend issue. https://github.com/kubeflow/pipelines/pull/766

Reply:
Yes. In this case, the containers running as steps of the pipeline are storing the data in the object store. The UI then visualizes the result. Only GCP GCS is supported at this time and we should support additional object stores.